### PR TITLE
Add GoroutinesV2 diagnostic handler for emitting goroutine pprofs

### PR DESCRIPTION
**Pull request by jamesross3**

### CODE REVIEW

**Comment by jamesross3** [comment]
@nmiyake and @bmoylan - I've hit a wall using the existing goroutine pprof diagnostic when debugging goroutine leaks. The main issue is that some runtime/pprof features such as goroutine labeling don't make it into the txt output which this existing diagnostic produces. I propose we add this pprof output for goroutines, which enables expedited goroutine leak detection. See https://rakyll.org/profiler-labels/ for a simple example of goroutine pprof visualizations with labels.

File: witchcraft/internal/wdebug/handlers.go [hunk 1]
```diff
@@ -76,7 +78,7 @@ func (h handlerGoroutinesV1) SafeLoggable() bool {
 }
 
 func (h handlerGoroutinesV1) Extension() string {
-	return "txt"
+	return "prof"
```
Comment (bmoylan) [inline witchcraft/internal/wdebug/handlers.go]: Is this useful somehow on the old format or can we keep it as-is?

Comment (jamesross3) [reply witchcraft/internal/wdebug/handlers.go]: oops meant to revert this

Comment (jamesross3) [reply witchcraft/internal/wdebug/handlers.go]: fixed

File: changelog/@unreleased/pr-850.v2.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+type: feature
+feature:
+  description: |
+    Add DiagnosticTypeGoroutinesV2 debug handler.
+  
+    This diagnostic emits a goroutines pprof, compatible with go tool pprof. It provides additional information which the existing DiagnosticTypeGoroutinesV1 does not, such as pprof.LabelSet attribution of goroutines.
+  links:
+    - https://github.com/palantir/witchcraft-go-server/pull/850
```
Comment (nmiyake) [inline changelog/@unreleased/pr-850.v2.yml]: nit: add newline at end of file

Comment (jamesross3) [reply changelog/@unreleased/pr-850.v2.yml]: Thanks - done

**Review by bmoylan — APPROVED** [review]

**Pull request merged**